cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1662,6 +1694,38 @@ func (p *BicepProvider) compileBicep(
 	}, nil
 }
 
+func combineMetadata(base map[string]json.RawMessage, override map[string]json.RawMessage) map[string]json.RawMessage {
+	if base == nil && override == nil {
+		return nil
+	}
+
+	if override == nil {
+		return base
+	}
+
+	// final map is expected to be at lease the same size as the base
+	finalMetadata := make(map[string]json.RawMessage, len(base))
+
+	for key, data := range base {
+		finalMetadata[key] = data
+	}
+
+	for key, data := range override {
+		finalMetadata[key] = data
+	}
+
+	return finalMetadata
+}
+
+func definitionName(typeDefinitionRef string) (string, error) {
+	definitionKeyNameTokens := strings.Split(typeDefinitionRef, "/")
```

Comment:
Just checking - I am guessing that forward slash is outside the domain of the allowable characters in a name.  If not, perhaps `strings.SplitN` would be better so we can constraint the returned slice to have at most two elements?  Or if there are multiple `/`'s do we really want to take the last piece?

Reply:
> forward slash is outside the domain of the allowable characters in a name

Yes, that would be the name of the user-defined-type, which follows the same naming rules as variables and parameters for bicep language.

A reference for a UDT would typically look like: `"$ref": "#/definitions/userDefinedTypeName"`
I need to pull the `userDefinedTypeName` out of that string... 
Initially I thought about a simple and straightforward solution by finding and removing the constant `#/definitions/`, but I felt like it could be easily break if someone ever decide to use a different path within the json for the `definitions` or if they introduce some organization levels between the actual definition name and the root node.
So, I felt better splitting and taking the last value from the slice.

I also looked a little bit a way to make json.marshaller to automatically follow the `reference` field and resolve the actual values based on the reference. I think the reference stuff is a json-standard approach, so I though there must be a way to ask the marshaller to make sure the reference exists and do the resolution...  But I gave up after some searching.. XD  .   Let me know if you know it this is possible!!

Reply:
That makse a ton of sense. This is fine for now without implementing / getting a library support with [JSON Pointers](https://datatracker.ietf.org/doc/html/rfc6901).

Could be nice to just add a concrete example above `strings.Split`:
```
// We typically expect `#/definitions/<name>` or `/definitions/<name>`, but loosely, we simply take `<name>` as the last separated element.
```

Reply:
updated